dsl/src/test/scala/org/http4s/dsl/PathSpec.scala

```diff
@@ -88,84 +84,58 @@ class PathSpec extends Http4sSpec {
       }) must beTrue
     }
 
-    "Root extractor" in {
-      (Path("/") match {
-        case Root => true
-        case _ => false
-      }) must beTrue
-    }
-
-    "Root extractor, no partial match" in {
-      (Path("/test.json") match {
-        case Root => true
-        case _ => false
-      }) must beFalse
-    }
-
-    "Root extractor, empty path" in {
-      (Path("") match {
-        case Root => true
-        case _ => false
-      }) must beTrue
-    }
-
     "/ extractor" in {
-      (Path("/1/2/3/test.json") match {
+      (Path.fromString("/1/2/3/test.json") match {
         case Root / "1" / "2" / "3" / "test.json" => true
         case _ => false
       }) must beTrue
     }
 
     "/: extractor" in {
-      (Path("/1/2/3/test.json") match {
+      (Path.fromString("/1/2/3/test.json") match {
         case "1" /: "2" /: path => Some(path)
         case _ => None
-      }) must_== Some(Path("/3/test.json"))
+      }) must_== Some(Path.fromString("3/test.json"))
     }
 
     "/: should not crash without trailing slash" in {
       // Bug reported on Gitter
-      Path("/cameras/1NJDOI") match {
+      Path(Vector(Segment("/cameras/1NJDOI"))) match {
         case "cameras" /: _ /: "events" /: _ /: "exports" /: _ => false
         case _ => true
       }
     }
 
     "trailing slash" in {
-      (Path("/1/2/3/") match {
+      (Path.fromString("/1/2/3/") match {
         case Root / "1" / "2" / "3" / "" => true
         case _ => false
       }) must beTrue
     }
 
     "encoded chars" in {
-      (Path("/foo%20bar/and%2For/1%2F2") match {
+      (Path.fromString("/foo%20bar/and%2For/1%2F2") match {
         case Root / "foo bar" / "and/or" / "1/2" => true
         case _ => false
       }) must beTrue
     }
 
-    "encode chars in toString" in {
-      (Root / "foo bar" / "and/or" / "1/2").toString must_==
-        "/foo%20bar/and%2For/1%2F2"
-    }
-
     "Int extractor" in {
-      (Path("/user/123") match {
+      (Path.fromString("/user/123") match {
         case Root / "user" / IntVar(userId) => userId == 123
         case _ => false
       }) must beTrue
     }
 
     "Int extractor, invalid int" in {
-      (Path("/user/invalid") match {
+      (Path(Vector(Segment("/user/invalid"))) match {
```

Comment:
This cant be right? 
`Path.fromString` should be used here?

Reply:
done